Code-4-Community/constellation
# Amplify Basic Setup

### ℹ️ Issue

Closes #4 

### 📝 Description

Briefly list the changes made to the code:

- Added basic Amplify boilerplate code following [this tutorial](https://ui.docs.amplify.aws/react/connected-components/authenticator)
- Created a dummy authenticated page component on the frontend

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

Setup steps:
1. Manually configure a Cognito userpool for constellation using the AWS console
2. Manually create a user within that userpool in the AWS console
3. Run `amplify pull`
4. Run `npm install`
5. Run `nx serve frontend`
6. Navigate to `localhost:4200` in your browser
7. Sign in!

<img width="736" alt="image" src="https://user-images.githubusercontent.com/55663526/229255940-6587f5b2-415e-4437-aa0b-fe4c5ddbab94.png">

<img width="1166" alt="image" src="https://user-images.githubusercontent.com/55663526/229255955-94009a94-8b6b-42b3-88c5-27f8ea9d1915.png">



### 🏕️ (Optional) Future Work / Notes

@bellehlim or @jackielincroft would you guys make tickets for these and assign them?

[ ] Make the signup page prettier - it's kinda ugly rn
[ ] Remove the option to Create Account in the first screenshot - this might require some customization of the `withAuthenticator` wrapper around `App` or removing that wrapper entirely and using more custom UI components
[ ] (Optional, only if someone's interested) Add Cognito userpool setup to our AWS SAM `template.yaml` so we don't need to rely on manual infrastructure setup steps
